driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java

```diff
@@ -0,0 +1,589 @@
+            stmt.addBatch();
+
+            stmt.setInt(1, 4);
+            stmt.setString(2, "baz");
+            stmt.addBatch();
+
+            try {
+                stmt.executeBatch();
+                fail("BatchUpdateException not thrown");
+            } catch (BatchUpdateException e) {
+                assertArrayEquals(
+                    new int[]{Statement.EXECUTE_FAILED, Statement.EXECUTE_FAILED, Statement.EXECUTE_FAILED},
+                    e.getUpdateCounts()
+                );
+            }
+            conn.createStatement().execute("refresh table test");
+            ResultSet resultSet = conn.createStatement().executeQuery("select count(*) from test");
+            assertThat(resultSet.next(), is(true));
+            assertThat(resultSet.getLong(1), is(1L));
+        }
+    }
+
+    @Test
+    public void testTypesResponseNoResult() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            ResultSet result = conn.createStatement().executeQuery("select * from test where 1=0");
+            ResultSetMetaData metaData = result.getMetaData();
+            assertThat(metaData.getColumnCount(), is(12));
+            for (int i = 1; i <= result.getMetaData().getColumnCount(); i++) {
+                // test that we can get the types, whatever they are
+                assertThat(metaData.getColumnType(i), instanceOf(Integer.class));
+            }
+        }
+    }
+
+    @Test
+    public void testSelectWhenNothingMatches() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            assertTrue(conn.createStatement().execute("select * from test where string_field = 'nothing_matches_this'"));
+        }
+    }
+
+    @Test
+    public void testExecuteUpdateWhenNothingMatches() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            assertThat(conn.createStatement().executeUpdate("update test set string_field = 'new_value'" +
+                " where string_field = 'nothing_matches_this'"), is(0));
+        }
+    }
+
+
+    @Test
+    public void testMultipleHostsConnectionString() throws Exception {
+        CrateTestServer server = testCluster.randomServer();
+        String connectionStr = String.format(
+                "crate://%s:%s,%s:%s/", server.crateHost(), server.psqlPort(), server.crateHost(), server.psqlPort()
+        );
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            assertThat(conn.createStatement().execute("select 1 from sys.cluster"), is(true));
+        }
+    }
+
+    @Test
+    @Ignore("getSchemas is not implemented")
+    public void testGetSchemas() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            DatabaseMetaData metaData = conn.getMetaData();
+            ResultSet result = metaData.getSchemas();
+            result.next();
```

Comment:
pls test that getSchemas returns valid schemas only. testing for all implemented schemas is imho beyond of this test and will fail if other schemas are implemented by the server